RFCs/FS-1090-Generic-struct-type-whose-fields-are-all-unmanaged-types-is-unmanaged.md

```diff
@@ -0,0 +1,198 @@
+# F# RFC FS-1088 - (Generic struct type whose fields are all unmanaged types is unmanaged)
+
+The design suggestion [Generic struct type whose fields are all unmanaged types is unmanaged](https://github.com/fsharp/fslang-suggestions/issues/692) has been marked "approved in principle".
+This RFC covers the detailed proposal for this suggestion.
+
+* [x] Approved in principle
+* [x] [Suggestion](https://github.com/fsharp/fslang-suggestions/issues/692)
+* [ ] Implementation: [In progress](https://github.com/dotnet/fsharp/pull/6064)
+
+# Summary
+
+[summary]: #summary
+
+Allow generic structs to be `unmanaged` at construction if all fields are `unmanaged` types.
+
+# Motivation
+
+[motivation]: #motivation
+
+* Make it easier to author low level interop code.
+* Improve interoperability with C#.
+
+# Detailed design
+
+[design]: #detailed-design
+
+Currently, it is not possible to have an `unmanaged` constraint working on generic struct types.
+
+Consider following code:
+
+```fsharp
+[<Struct>]
+type Test<'T when 'T: unmanaged> =
+        val element: 'T
+
+let works = Test<int>()
+let error = Test<Test<int>>()
+
+```
+
+Prior to this RFC, the example above will fail to compile with:
+
+```less
+let error = Test<Test<int>>()
+  ------------^^^^^^^^^^^^^^^
+
+stdin(6,13): error FS0001: A generic construct requires that the type 'Test<int>' is an unmanaged type
+```
+
+This proposal aims to eliminate this restriction by treating generic struct type `unmanaged` if all its fields are `unmanaged`.
```

Comment:
```suggestion
This proposal aims to resolve this inconsistency by treating a generic struct type as `unmanaged` if all its fields are unmanaged.
```

Reply:
The  way this is worded makes it sound like I won't need to specify `unmanaged` in a generic struct definition, and it will "just work" so long as the `'T` being substituted is unamanaged. Is that correct, or do I need to be explicit in my type definition?

Reply:
@cartermp Yeah, the idea is that you won't need to specify the unmanaged constraint and it will "just work".
I will add a note about it in the RFC.